src/annotation.rs

```diff
@@ -222,11 +223,52 @@ pub enum Max<K> {
     Maximum(K),
 }
 
+impl<K> PartialEq<K> for Max<K>
+where
+    K: PartialOrd,
+{
+    fn eq(&self, rhs: &K) -> bool {
+        match (self, rhs) {
+            (Max::Maximum(k), rhs) => k == rhs,
+            _ => false,
+        }
+    }
+}
+
+impl<K> PartialOrd<K> for Max<K>
+where
+    K: PartialOrd,
+{
+    fn partial_cmp(&self, other: &K) -> Option<Ordering> {
+        match self {
+            Max::NegativeInfinity => Some(Ordering::Less),
+            Max::Maximum(m) => m.partial_cmp(other),
+        }
+    }
+}
+
+impl<K> PartialOrd for Max<K>
+where
+    K: PartialOrd,
+{
+    fn partial_cmp(&self, other: &Max<K>) -> Option<Ordering> {
+        // Prevent ordering inconsistency for cmp between two negative infinity
+        if self == other {
+            return Some(Ordering::Equal);
+        }
+
+        match other {
+            Max::NegativeInfinity => Some(Ordering::Greater),
+            Max::Maximum(other) => self.partial_cmp(other),
+        }
+    }
+}
+
 impl<C, S, K> Annotation<C, S> for Max<K>
 where
     C: Compound<S>,
     S: Store,
-    K: Ord + Clone,
+    K: PartialOrd + Clone,
```

Comment:
Is it really needed to force that? Since this constraint will have implications in the rest of annotation impls we have in other repos such as the PoseidonAnnotation no?

Reply:
Yes, the `Max` annotation requires the key to be `PartialOrd`